airflow/contrib/operators/pubsub_operator.py

```diff
@@ -0,0 +1,162 @@
+    ```
+    with DAG('successful DAG') as dag:
+        (
+            dag
+            >> PubSubTopicCreateOperator(topic='my_new_topic')
+            >> PubSubTopicCreateOperator(topic='my_new_topic')
+        )
+    ```
+
+    The operator can be configured to fail if the topic already exists.
+    ```
+    with DAG('failing DAG') as dag:
+        (
+            dag
+            >> PubSubTopicCreateOperator(topic='my_new_topic')
+            >> PubSubTopicCreateOperator(topic='my_new_topic',
+                                         fail_if_exists=True)
+        )
+    ```
+    """
+    template_fields = []
+    ui_color = '#0273d4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            project,
+            topic,
+            fail_if_exists=False,
+            gcp_conn_id='google_cloud_default',
+            delegate_to=None,
+            *args,
+            **kwargs):
+        """
+        :param project: the GCP project name or ID in which to work
+        :type project: string
+        :param topic: the topic to create. Do not include the
+            full topic path. In other words, instead of
+            ``projects/{project}/topics/{topic}``, provide only
+            ``{topic}``.
+        :type topic: string
+        :param gcp_conn_id: The connection ID to use connecting to
+            Google Cloud Platform.
+        :type gcp_conn_id: string
+        :param delegate_to: The account to impersonate, if any.
+            For this to work, the service account making the request
+            must have domain-wide delegation enabled.
+        :type delegate_to: string
+        """
+        super(PubSubTopicCreateOperator, self).__init__(*args, **kwargs)
+
+        self.project = project
+        self.topic = topic
+        self.fail_if_exists = fail_if_exists
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+
+    def execute(self, context):
+        hook = PubSubHook(gcp_conn_id=self.gcp_conn_id,
+                          delegate_to=self.delegate_to)
+
+        hook.create_topic(self.project, self.topic,
+                          fail_if_exists=self.fail_if_exists)
+
+
+class PubSubPublishOperator(BaseOperator):
+    """Publish messages to a PubSub topic.
+
+    Each Task publishes all provided messages to the same topic
+    in a single GCP project. If the topic does not exist, this
+    task will fail.
+
+    ```
+    from base64 import b64encode as b64e
+
+    m1 = {'data': b64e('Hello, World!'),
+          'attributes': {'type': 'greeting'}
+         }
+    m2 = {'data': b64e('Knock, knock')}
+    m3 = {'attributes': {'foo': ''}}
+
+    t1 = PubSubPublishOperator(
+        topic='my_topic',
+        messages=[m1, m2, m3],
+        create_topic=True,
+        dag=dag)
+    ```
+    """
+    template_fields = []
+    ui_color = '#0273d4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            project,
+            topic,
+            messages,
```

Comment:
It does work, I used it somewhere before. Indeed, add it to the templated fields as well.